src/main/java/com/rabbitmq/client/impl/recovery/TopologyRecoveryRetryLogic.java

```diff
@@ -138,18 +149,52 @@
      * Recover a consumer.
      */
     public static final DefaultRetryHandler.RetryOperation<String> RECOVER_CONSUMER = context -> context.consumer().recover();
+    
+    /**
+     * Recover earlier consumers that share the same channel as this retry context
+     */
+    public static final DefaultRetryHandler.RetryOperation<String> RECOVER_PREVIOUS_CONSUMERS = context -> {
+        if (context.entity() instanceof RecordedConsumer) {
+            // recover all consumers for the same channel that were recovered before this current
+            // consumer. need to do this incase some consumers had already been recovered
+            // successfully on a different queue before this one failed
+            final AutorecoveringChannel channel = context.consumer().getChannel();
+            for (RecordedConsumer consumer : Utility.copy(context.connection().getRecordedConsumers()).values()) {
+                if (consumer == context.entity()) {
+                    break;
+                } else if (consumer.getChannel() == channel) {
+                    final RetryContext retryContext = new RetryContext(consumer, context.exception(), context.connection());
+                    RECOVER_CONSUMER_QUEUE.call(retryContext);
+                    consumer.recover();
+                    RECOVER_CONSUMER_QUEUE_BINDINGS.call(retryContext);
+                }
+            }
+        }
+        return null;
+    };
 
     /**
      * Pre-configured {@link TopologyRecoveryRetryHandlerBuilder} that retries recovery of bindings and consumers
      * when their respective queue is not found.
+     * 
      * This retry handler can be useful for long recovery processes, whereby auto-delete queues
      * can be deleted between queue recovery and binding/consumer recovery.
+     * 
+     * Also useful to retry channel-closed 404 errors that may arise with auto-delete queues during a cluster cycle.
      */
     public static final TopologyRecoveryRetryHandlerBuilder RETRY_ON_QUEUE_NOT_FOUND_RETRY_HANDLER = builder()
+        .queueRecoveryRetryCondition(CHANNEL_CLOSED_NOT_FOUND)
         .bindingRecoveryRetryCondition(CHANNEL_CLOSED_NOT_FOUND)
         .consumerRecoveryRetryCondition(CHANNEL_CLOSED_NOT_FOUND)
-        .bindingRecoveryRetryOperation(RECOVER_CHANNEL.andThen(RECOVER_BINDING_QUEUE).andThen(RECOVER_BINDING)
+        .queueRecoveryRetryOperation(RECOVER_CHANNEL
```

Comment:
Have seen channel-closed 404 "process is stopped by supervisor" errors during the recovery of a queue as well.